### PR TITLE
CC-981 Support toggling concept status

### PR DESCRIPTION
**Pull request by libmartinito**
**Checklist**:

- [x] I've thoroughly self-reviewed my changes
- [x] I've added tests for my changes, unless they affect admin-only areas (or are otherwise not worth testing)
- [ ] I've verified any visual changes using Percy (add a commit with `[percy]` in the message to trigger)

## Summary by CodeRabbit

- **New Features**
  - Introduced a "Draft" label with a tooltip for draft concepts visible only to staff.
  - Added a toggle switch to control concept publishing status from the "Basic Details" admin page.

- **Enhancements**
  - Improved concept visibility filtering to show only relevant concepts based on user roles and statuses.

- **Tests**
  - Added test cases to verify the correct toggling of concept publish status and the visibility of draft labels for both draft and published concepts.

### CODE REVIEW

File: app/components/concepts-page/concept-card.hbs [hunk 1]
```diff
@@ -13,6 +13,13 @@
       {{#if (current-user-is-concept-author @concept)}}
         <Pill @color="green">Yours</Pill>
       {{/if}}
+
+      {{#if (eq @concept.status "draft")}}
+        <Pill @color="gray" data-test-draft-label>
+          Draft
+          <EmberTooltip @text="This concept is only visible to staff" />
+        </Pill>
+      {{/if}}
```
Comment (libmartinito) [inline app/components/concepts-page/concept-card.hbs:22]: feels like the two pills communicate something different, so added both

File: app/controllers/concepts.ts [hunk 2]
```diff
@@ -18,12 +19,24 @@ export default class ConceptsController extends Controller {
   @service declare router: RouterService;
   @service declare authenticator: AuthenticatorService;
 
+  get currentUser(): UserModel | null {
```
Comment (rohitpaulk) [inline app/controllers/concepts.ts:22]: If the return type says `UserModel | null`, why are we returning with `as UserModel`?

File: app/templates/concept-admin/basic-details.hbs [hunk 3]
```diff
@@ -103,6 +103,28 @@
...
+            Unpublished concepts aren't visible to other users.
+          </label>
+        </:description>
+
+        <:content>
+          <Toggle @isOn={{this.isConceptPublished}} {{on "click" this.handlePublishConceptToggled}} class="mt-1" data-test-publish-concept-toggle />
+
+          {{#if this.isConceptPublished}}
+            <div class="text-gray-400 text-xs mt-0.5">
+              This concept is published
```
Comment (rohitpaulk) [inline app/templates/concept-admin/basic-details.hbs:119]: If a concept was `published` already, this would show a wrong value

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Added notes

**Review by rohitpaulk — APPROVED** [review]

**Pull request merged**